pages/docs/styled-system/theming/theme-tools.mdx

```diff
@@ -0,0 +1,233 @@
+the range values) (see the [default theme colors](../theming/theme#colors))
+
+```js
+const theme = extendTheme({
+  components: {
+    Button: {
+      variants: {
+        variantCustom: (props) => {
+          const { theme, colorScheme: c } = props
+          // Some setup logic here
+          return {
+            // Apply styles here
+          }
+        },
+      },
+    },
+  },
+})
+```
+
+> Note: This is where it's best to split up your theme file to
+> [scale out your project](../theming/customize-theme#scaling-out-your-project)
+
+## Style Function Props
+
+When creating a variant or base style with dynamic styles, it's value can be a
+function instead of an object to use props that are passed to the component. The
+following props can be used as part of the signature:
+
+| Prop          | Type / Value                                                                                                   |
+| ------------- | -------------------------------------------------------------------------------------------------------------- |
+| `colorScheme` | string value of a theme token (ie. `blue` or `green`)                                                          |
+| `theme`       | The theme object                                                                                               |
+| `colorMode`   | `Dark` or `Light` depending on the current value in the ColorMode Provider                                     |
+| `orientation` | `horizontal` or `vertical`, defined as the direction the component is laid out. (Like setting `flexDirection`) |
+
+And any other props passed to the component you want to access!
+
+## Colors
+
+Below are the methods that can be used to access and alter the colors from the
+theme.
+
+| Name                 | Description                                                                                                                                                                                                                                                                                          | Function Call                                                         |
+| -------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------- |
+| `getColor`           | Retrieves a color hex value. Needs the theme and color name, and can take a fallback hex if the color name does not return a valid hex from the theme.                                                                                                                                               | `getColor(theme, color, fallback)`                                    |

```

Comment:
After reading the result in the preview multiple time, I find this table hard to read and understand. What about just doing header with the function name and sections with the informations as you did for mode()

Reply:
Yeah you can add a note :) I commented it in the actual code but maybe two places would help